spirv_glsl.cpp

```diff
@@ -5331,6 +5542,32 @@ void CompilerGLSL::require_extension(const string &ext)
 	}
 }
 
+void CompilerGLSL::flatten_buffer_block(uint32_t id)
+{
+	auto &var = get<SPIRVariable>(id);
+	auto &type = get<SPIRType>(var.basetype);
+	auto name = to_name(type.self, false);
+	auto flags = meta.at(type.self).decoration.decoration_flags;
+
+	if (!type.array.empty())
+		SPIRV_CROSS_THROW(name + " is an array of UBOs.");
+	if (type.basetype != SPIRType::Struct)
+		SPIRV_CROSS_THROW(name + " is not a struct.");
+	if ((flags & (1ull << DecorationBlock)) == 0)
+		SPIRV_CROSS_THROW(name + " is not a block.");
+	if (type.member_types.empty())
+		SPIRV_CROSS_THROW(name + " is an empty struct.");
+
+	uint64_t member_flags = 0;
+	for (uint32_t i = 0; i < type.member_types.size(); i++)
+		member_flags |= combined_decoration_for_member(type, i);
+
+	if (member_flags & (1ull << DecorationRowMajor))
```

Comment:
We could probably support this if the access chain terminates at the matrix, the access chain expression can just get "need_transpose" in that case.